File: scripts/test_sum/test_sum.js
//the test case (js)
function test_sum(key, val) {
    function sum(n) {
        if (n < 2) {
            return n;
        }

        return n + sum(n - 1);
    }

    const result = sum(val);
    console.log(`${key}: ${result}`);
}

for (let i = 0; i <= 10; i++) {
    test_sum(i, i * 1000);
}

File: scripts/test_sum/test_sum.c
#include <stdio.h>

//functions be local in C
int sum(int n) {
	if (n < 2) {
		return n;
	}

	return n + sum(n - 1);
}

//the test case (C)
void test_sum(int key, int val) {
	const int result = sum(val);
	printf("%d: %d\n", key, result);
}

int main() {
	for (int i = 0; i <= 10; i++) {
		test_sum(i, i * 1000);
	}
}